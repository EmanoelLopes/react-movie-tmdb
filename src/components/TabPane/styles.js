import styled from 'styled-components';
import { colors } from 'utils/styled';

export const StyledTabPane = styled.div`
  cursor: pointer;
  display: flex;
  width: 50%;

  &.tab-active {
    color: ${colors.movieDBColor};
  }
`;
