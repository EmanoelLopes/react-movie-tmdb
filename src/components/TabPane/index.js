import React from 'react';
import {
  bool,
  func,
  number,
  string,
  node
} from 'prop-types';
import { H1 } from 'utils/styled';
import { StyledTabPane } from './styles';

const TabPane = props => {
  const { label, index, isActive, icon, onClick } = props;
  
  return (
    <StyledTabPane
      index={index}
      onClick={onClick}
      className={`tab-pane ${(isActive) ? 'tab-active' : ''}`}>
      <H1>
        <span className="label">{label}</span> {icon}
      </H1>
    </StyledTabPane>
  );
};

TabPane.propTypes = {
  label: string.isRequired,
  onClick: func.isRequired,
  index: number.isRequired,
  isActive: bool.isRequired,
  icon: node.isRequired,
};

export default TabPane;
