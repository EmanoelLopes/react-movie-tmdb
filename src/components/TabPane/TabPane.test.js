import React from 'react';
import { mount } from 'enzyme';
import TabPane from 'components/TabPane';

let wrapped;
let defaultProps = {
  label: 'Label',
  index: 0,
  isActive: true,
  icon: '',
  onClick: () => null,
};

beforeEach(() => {
  wrapped = mount(<TabPane {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<TabPane />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
