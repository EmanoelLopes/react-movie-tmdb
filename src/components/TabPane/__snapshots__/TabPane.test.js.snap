// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPane /> should match snapshot 1`] = `
.c1 {
  font-size: 32px;
  margin: 25px 0;
  width: 100%;
  text-transform: uppercase;
}

.c0 {
  background-color: #fff;
  border-bottom: 5px solid #fff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 15px 0;
  text-align: center;
  width: 50%;
}

.c0.tab-active {
  color: #01d277;
  border-bottom: 5px solid #01d277;
}

.c0 .label {
  display: none;
}

@media (min-width:36em) {
  .c0 .label {
    display: initial;
  }
}

<TabPane
  icon=""
  index={0}
  isActive={true}
  label="Label"
  onClick={[Function]}
>
  <styled.div
    className="tab-pane tab-active"
    index={0}
    onClick={[Function]}
  >
    <StyledComponent
      className="tab-pane tab-active"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  background-color: ",
              "#fff",
              ";
  border-bottom: 5px solid ",
              "#fff",
              ";
  cursor: pointer;
  display: flex;
  margin: 15px 0; 
  text-align: center;
  width: 50%;

  &.tab-active {
    color: ",
              "#01d277",
              ";
    border-bottom: 5px solid ",
              "#01d277",
              ";
  }

  .label {
    display: none;

    ",
              "
    @media (min-width: ",
              "36",
              "em) {
      ",
              "
      display: initial;
    ",
              "
    }
  ",
              "
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      index={0}
      onClick={[Function]}
    >
      <div
        className="tab-pane tab-active c0"
        onClick={[Function]}
      >
        <styled.h1>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  font-size: 32px;
  margin: 25px 0;
  width: 100%;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.h1",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="c1"
            >
              <span
                className="label"
              >
                Label
              </span>
               
            </h1>
          </StyledComponent>
        </styled.h1>
      </div>
    </StyledComponent>
  </styled.div>
</TabPane>
`;
